doc/adr/0007-queryable-properties.md

```diff
@@ -0,0 +1,286 @@
+- User can click on a callsite and source code file will be opened using IDE's capabilities
+
+### Example Sequence Diagram
+
+Prerequiste `Open Trace and fetched property descriptors` for all diagrams:
+
+```mermaid
+sequenceDiagram
+    actor User
+    participant client as Theia Trace
+    participant server as Trace Server
+    User->>client: open an experiment
+    client->>server: POST /experiments/
+    server->>client: 200: Experiment
+    client->>server: GET /experiment/{expUuid}/outputs
+    server->>client: 200: OutputDescriptor[]
+    Note over client,server: OutputDescriptor contains list of PropertyDescriptor
+   
+```
+
+#### Go To Max
+
+The following feature exists in Eclipse Trace Compass for all latency statistics tables. It provides the possibility to navigate to the time range where the maximum latency occured. The metadata give some additional information that can be used in the front-end to identify UI elements (not available in Eclipse Trace Compass).
+
+```mermaid
+sequenceDiagram
+    actor User
+    participant client as Theia Trace
+    participant server as Trace Server
+    Note over client,server: Prerequisite Open Trace and fetched property descriptors
+    User->>client: open Latency Table
+    client->>server: POST /experiment/{expUuid}/outputs/{outputId}/data/tree
+    server->>client: 200: TreeModel
+    client->>client: populate view
+    User->>client: open view menu
+    loop while descriptor in propertyDescriptors
+        client->>client: create menu entry (using descriptor.name, descriptor.description, descriptor.propertyId)
+    end
+    User->>client: select "Go To Max" menu item
+    client->>server: POST /experiment/{expUuid}/outputs/{outputId}/data/properties/{propertyId}
+    Note over client,server: requested_items=[selected_entry_Id]
+    server->>client: PropertyModel ([ { time_range: {start, end}, metatdata: (cpu, 0) } ])
+    client->>client: broadcast selectionRange({ time_range:{start, end}, metatdata: (cpu, 0) })
+    client->>client: in all views, update selection range and show cpu (if applicable) 
+    User->>client: navigate to selectionRange {start, end}
+```
+
+#### Get 10 longest executions
+
+This features could be used on latency tables to provide the top 10 latencies for a function.
+
+```mermaid
+sequenceDiagram
+    actor User
+    participant client as Theia Trace
+    participant server as Trace Server
+    Note over client,server: Prerequisite Open Trace and fetched property descriptors
+    User->>client: open Latency Table
+    client->>server: POST /experiment/{expUuid}/outputs/{outputId}/data/tree
+    server->>client: 200: TreeModel
+    client->>client: populate view
+    User->>client: open view menu
+    loop while descriptor in propertyDescriptors
+        client->>client: create menu entry (using descriptor.name, descriptor.description, descriptor.propertyId)
+    end
+    User->>client: select "Go To Max" menu item
```

Comment:
I think here "Go to max" is a copy/paste error

Reply:
yep, copy paste error.